File: server/utils/fetch_news.py
from fastapi import HTTPException
import requests
from constants import *
from utils.articler_parser import parse_news_data_to_articles


def fetch_news(topic: str) -> list:
    try:
        # Make a request to the News API
        response = requests.get(f"{base_url}{topic}{api_key}")
        response.raise_for_status()  # Raise an exception if request fails
        news_data = response.json()
        news_articles = news_data.get("articles","Error occured while procuring data")
        news_articles = parse_news_data_to_articles(news_articles)
        return news_articles
    
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

File: src/utils/fetch_news.js
import axios from "axios";
import { LOCAL_URL } from "../constants/constants";

export const FetchNews = async (topic) => {
  const config = {
    method: "get",
    maxBodyLength: Infinity,
    url: `${LOCAL_URL}news/${topic}`,
    headers: {},
  };

  const response = await axios.request(config);
  if (response.status === 200) {
    return response.data;
  } else {
    throw new Error("Error fetching news");
  }
};
